Apps/W1/Shopify/test/Logs/ShpfySkippedRecordLogTest.Codeunit.al

```diff
@@ -0,0 +1,838 @@
+codeunit 139581 "Shpfy Skipped Record Log Test"
+{
+    Subtype = Test;
+    TestPermissions = Disabled;
+
+    var
+        Shop: Record "Shpfy Shop";
+        ShpfyInitializeTest: Codeunit "Shpfy Initialize Test";
+        LibraryAssert: Codeunit "Library Assert";
+        Any: Codeunit Any;
+        SalesShipmentNo: Code[20];
+        IsInitialized: Boolean;
+
+    trigger OnRun()
+    begin
+        IsInitialized := false;
+    end;
+
+    [Test]
+    procedure UnitTestLogEmptyCustomerEmail()
+    var
+
+        Customer: Record Customer;
+        SkippedRecord: Record "Shpfy Skipped Record";
+        ShpfyCustomerExport: Codeunit "Shpfy Customer Export";
+        ShpfySkippedRecordLogSub: Codeunit "Shpfy Skipped Record Log Sub.";
+    begin
+        // [SCENARIO] Log skipped record when customer email is empty on customer export to shopify.
+        Initialize();
+
+        // [GIVEN] A customer record with empty email.
+        CreateCustomerWithEmail(Customer, '');
+
+        // [WHEN] Invoke Shopify Customer Export
+        BindSubscription(ShpfySkippedRecordLogSub);
```

Comment:
A lot of these comments are going to be Test Automation preferences. If MS has been writing tests the same way you wrote them, I'm fine with this different approach.

There's too much happening in this WHEN step. It takes away from the readability and would be better off extracted.